src/libraries/context-recording-selection/RecordingSelectionContext.ts

```diff
@@ -346,6 +363,40 @@ const setFocusTime = (state: RecordingSelection, action: SetFocusTimeRecordingSe
             }
         }
     }
+    if (shiftKey) {
+        const t0 = state.focusTimeSeconds
+        if (t0 !== undefined) {
+            const t1 = Math.min(t0, focusTimeSec)
+            const t2 = Math.max(t0, focusTimeSec)
+            newState = {...newState, focusTimeSeconds: state.focusTimeSeconds, focusTimeIntervalSeconds: [t1, t2]}
+        }
+    }
+    return selectionIsValid(newState) ? newState : state
+}
+
+const setFocusTimeInterval = (state: RecordingSelection, action: SetFocusTimeIntervalRecordingSelectionAction): RecordingSelection => {
+    const {focusTimeIntervalSec, autoScrollVisibleTimeRange} = action
+    let newState: RecordingSelection = { ...state, focusTimeIntervalSeconds: focusTimeIntervalSec }
+    if (autoScrollVisibleTimeRange) {
+        const t0 = (action.focusTimeIntervalSec[0] + action.focusTimeIntervalSec[1]) / 2
+        if ((state.visibleTimeStartSeconds !== undefined) && (state.visibleTimeEndSeconds !== undefined)) {
+            if ((t0 < state.visibleTimeStartSeconds) || (t0 > state.visibleTimeEndSeconds)) {
+                const span = state.visibleTimeEndSeconds - state.visibleTimeStartSeconds
+                newState.visibleTimeStartSeconds = t0 - span / 2
+                newState.visibleTimeEndSeconds = t0 + span / 2
+                if (newState.visibleTimeEndSeconds > (state.recordingEndTimeSeconds || 0)) {
+                    const delta = (state.recordingEndTimeSeconds || 0) - newState.visibleTimeEndSeconds
+                    newState.visibleTimeStartSeconds += delta
+                    newState.visibleTimeEndSeconds += delta
+                }
+                if (newState.visibleTimeStartSeconds < (state.recordingStartTimeSeconds || 0)) {
+                    const delta = (state.recordingStartTimeSeconds || 0) - newState.visibleTimeStartSeconds
+                    newState.visibleTimeStartSeconds += delta
+                    newState.visibleTimeEndSeconds += delta
+                }
+            }
+        }
+    }
```

Comment:
Feels like I've seen this before somewhere, hmm... maybe I had to write something similar for the animation crop window?

"Set a window, with width equal to the current window's width, with time XYZ as the midpoint unless that puts the bounds outside the extant data, in which case move the window so it's entirely in bounds" seems like something that's sufficiently complex and likely to be done again that it could benefit from being a function. Maybe we should flag this as a low-priority issue to be addressed sometime.

Reply:
opening an issue